serde/src/lib.rs

```diff
@@ -44,23 +45,60 @@ impl fmt::Display for DeserializeError {
 
 impl Error for DeserializeError {}
 
-/// Size in bytes for a single `i8` in binary serialization.
-pub const I8_SIZE: usize = 1;
-/// Size in bytes for a single `u8` in binary serialization.
-pub const U8_SIZE: usize = 1;
-
-/// Maximum size in bytes for a single `i16` in binary serialization.
-pub const I16_MAX_SIZE: usize = (16 + 6) / 7;
-/// Maximum size in bytes for a single `u16` in binary serialization.
-pub const U16_MAX_SIZE: usize = (16 + 6) / 7;
-/// Maximum size in bytes for a single `i32` in binary serialization.
-pub const I32_MAX_SIZE: usize = (32 + 6) / 7;
-/// Maximum size in bytes for a single `u32` in binary serialization.
-pub const U32_MAX_SIZE: usize = (32 + 6) / 7;
-/// Maximum size in bytes for a single `i64` in binary serialization.
-pub const I64_MAX_SIZE: usize = (64 + 6) / 7;
-/// Maximum size in bytes for a single `u64` in binary serialization.
-pub const U64_MAX_SIZE: usize = (64 + 6) / 7;
+/// Types implementing this trait have a constant binary serialization size.
+///
+/// It can be `#[derive]`d using the [`SerializeSize`] derive macro.
+pub trait SerializeSize {
+    /// Size in bytes of the serialization.
+    const SIZE: usize;
+}
+/// Types implementing this trait have a maximum binary serialization size.
+///
+/// It can be `#[derive]`d using the [`SerializeMaxSize`] derive macro.
+pub trait SerializeMaxSize {
+    /// Maximum size in bytes of the serialization.
+    const MAX_SIZE: usize;
+}
+
+impl<T: SerializeSize> SerializeMaxSize for T {
+    const MAX_SIZE: usize = T::SIZE;
+}
+
+#[rustfmt::skip]
```

Comment:
Why are we skipping rustfmt?

Reply:
Because rustfmt wants to put the implementations on separate lines, making the repetitive code less clear IMO.